### PR TITLE
Homework for Lesson 6

### PR DESCRIPTION
**Pull request by filkodaria**
Homework_6: task_1, task_2, task_3.

### CODE REVIEW

File: hw_6/hw_6.1.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+*/
+
+function getEmployeeInfo(name) {
+	const employeeNames = ['Hanna', 'Alex', 'Bill', 'Jane', 'Bob'];
+	const employeeSalary = [2000, 3000, 7000, 1000, 3000];
+	const employeeInfo = [];
+
+	for (let i = 0; i <= employeeNames.length; i++) {
+		if (employeeNames[i] === name) {
+			employeeInfo.push(employeeNames[i], employeeSalary[i]);
```
Comment (anatoly-karpovich) [inline hw_6/hw_6.1.js]:
```suggestion
	const employeeIndex = employeeNames.indexOf(name)
			employeeInfo.push(employeeNames[employeeIndex], employeeSalary[employeeIndex]);
```
And add validation for index if it is -1 for example

File: hw_6/hw_6.2.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	for (let el of array) {
+		lengthArray.push(el.length);
+	}
+
+	maxLength = Math.max(...lengthArray);
+	let result = [];
+
+	for (let el of array) {
+		if (el.length === maxLength) {
+			result.push(el);
```
Comment (anatoly-karpovich) [inline hw_6/hw_6.2.js]:
```suggestion
	for(const word of array) {
if(word.length > result[0].length) {
result.length = 0
result[0] = word
} esle if (word.length === result[0].length) result.push(word)
}
```
And you'll just need to handle the first adding to results array like if(!result.length) result.push(word)

In this example there is no need to have 2 loops

File: hw_6/hw_6.3.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    и так пока в тексте не останется двух одинаковых букв стоящих рядом (через пробел и другой знак препинания можно)
+    Пример: aabc => bbc => cc => d
+*/
+
+// subtask 5
+function countSum(num) {
+	const transformedNum = num.toString().split('');
+	let count = 0;
+
+	for (let el of transformedNum) {
```
Comment (anatoly-karpovich) [inline hw_6/hw_6.3.js]:
```suggestion
	for (const el of transformedNum) {
```
Always use const in for..of loop

File: hw_6/hw_6.3.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		return count;
+	} else return countSum(count);
+}
+console.log(countSum(292));
+
+
+// subtask 6
+function shrinkString(string) {
+	const alphabet = 'abcdefghijklmnopqrstuvwxyz';
+	let result = string.toLowerCase().split('').sort();
```
Comment (anatoly-karpovich) [inline hw_6/hw_6.3.js]: 1. No need to sort, you need to find all pairs on the closest indexes. Like in "caaddc" you will change aa and dd, and won't change cc, couse they are not neighbours 
2. After changing letters you need to check array once again and etc
3. Dont use sort in such way without a callback

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

File: hw_6/hw_6.2.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+		if (!competitorPizza.includes(myList[i])) {
+			extraPizza.push(myList[i]);
+		}
+	}
+
+	if (extraPizza.length === 0) {
+		return null;
+	} else {
+		return extraPizza;
+	}
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js]:
```suggestion
return extraPizza.length ? extraPizza : null;
```

File: hw_6/hw_6.2.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+function getMaxLengthWord(string) {
+	const array = string.split(' ');
+	const lengthArray = [];
+
+	for (let el of array) {
+		lengthArray.push(el.length);
+	}
+
+	maxLength = Math.max(...lengthArray);
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js]: Explicit variable / constant declaration

File: hw_6/hw_6.2.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	let result = [];
+
+	for (let el of array) {
+		if (array.indexOf(el) === array.lastIndexOf(el)) {
+			result.push(el);
+		}
+	}
+	return result;
+}
+console.log(removeDuplicates(numbers));
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js:87]: Result output have to be `[ 2, 7, 20, 4, 5, 10 ]` with your solution I've got `[ 20, 5, 10 ]`

Just remove elements that occur in the array for the second time

File: hw_6/hw_6.2.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+
+// subtask 4
+const initalWord = 'Madam';
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js]:
```suggestion
const initialWord = 'Madam';
```

File: hw_6/hw_6.2.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+function isPalindrome(str) {
+	const array = str.toLowerCase().split('');
+	const arrayReversed = array.slice().reverse();
+
+	for (let i = 0; i <= array.length / 2; i++) {
+		if (array[i] !== arrayReversed[i]) {
+			return `\'${str}\' is not a palindrome.`
+		}
+	}
+	return `\'${str}\' is a palindrome.`
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js]:
```suggestion
			return `'${str}' is not a palindrome.`
		}
	}
	return `'${str}' is a palindrome.`
```

File: hw_6/hw_6.3.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+	while (result[0] === result[1]) {
+	const index = alphabet.indexOf(result[0]);
+	const indexOfNextAlphabet = index + 1 - alphabet.length;
+	result.splice(0, 2, alphabet.at(indexOfNextAlphabet));
+	}
+
+	return `Shrinked string is \'${result.toString()}\'.`
+}
+console.log(shrinkString('aabcde'));
```
Comment (VladimirLQA) [inline hw_6/hw_6.3.js]:
```suggestion
function shrinkString(string) {
  const alphabet = 'abcdefghijklmnopqrstuvwxyz';
  let result = string.toLowerCase().split('');
  while (result[0] === result[1]) {
    const replacementIdx = alphabet.indexOf(result[0]) + 1;
    result.splice(0, 2, alphabet.at(replacementIdx));
  }
  return `Shrinked string is '${result.toString()}'.`
}
console.log(shrinkString('aabcde'));
```

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: hw_6/hw_6.1.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+2. В функции создайте массив имен сотрудников 5шт (Имена придумать самим)
+3. В функции создайте массив зарплат сотрудников 5 шт(цифры придумать самим)
+4. Функция должна принимать 1 аргумент - имя сотрудника
+5. Функция должна возвращать новый массив, где первый элемент - имя сотрудника, второй - его зарплата
+6. Для поиска ответа функции нужно найти индекс сотрудника в массиве имен. Зарплату взять с ТЕМ ЖЕ индексом что и имя
+7. Для возврата из функции создайте массив, методом .push поместите в него имя и зарплату, и через return верните созданный массив
+8. Если такое имя сотрудника в массиве не найдется - вернуть null
+*/
+
+function getEmployeeInfo(name) {
```
Comment (VladimirLQA) [inline hw_6/hw_6.1.js:13]: Add validation that passed argument should be a string type, if not throw an error

File: hw_6/hw_6.2.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+3. Напишите функцию, которая принимает на вход массив чисел, убирает из него дубликаты и возвращает массив с только уникальными значениями.
+
+4. Написать функцию, которая принимает на вход слово и проверяет, является ли это слово палиндромом
+*/
+
+// subtask 1
+const competitorPizza = ['Pepperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'Hawaiian'];
+const pizzaList = ['Neapolitan', 'Margherita', 'Pepperoni', 'Hawaiian', 'BBQ Chicken', '4 cheeses'];
+
+function getPizzaComparison(myList) {
```
Comment (VladimirLQA) [inline hw_6/hw_6.2.js]: Add validation that passed argument should be an array, if not throw an error
Same for other functions ....

File: hw_6/hw_6.2.js (hunk 7 above)
Comment (filkodaria) [reply hw_6/hw_6.2.js:87]: Yeah, you're right!
maybe I haven't got the task requirement properly((

**Comment by filkodaria** [comment]
I've added changes to the files hw_6.1.js, hw_6.2.js, hw_6.3.js.
Please, check)

File: hw_6/hw_6.1.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+	if (typeof name !== 'string') {
+		return `Something went wrong! Only string type for entered data is valid.`
```
Comment (anatoly-karpovich) [inline hw_6/hw_6.1.js:15]:
```suggestion
		throw new Error(`Something went wrong! Only string type for entered data is valid.`)
```

**Review by anatoly-karpovich — APPROVED** [review]

**Review by VladimirLQA — APPROVED** [review]

**Comment by filkodaria** [comment]
Thanks a lot for your reviews, guys)

**Pull request merged**